torch/functional.py

```diff
@@ -722,3 +723,36 @@ def norm(input, p="fro", dim=None, keepdim=False, out=None):
         if out is None:
             return torch._C._VariableFunctions.norm(input, p, dim, keepdim=keepdim)
     return torch._C._VariableFunctions.norm(input, p, dim, keepdim=keepdim, out=out)
+
+
+def chain_matmul(*matrices):
+    r"""Returns the matrix product of the :math:`N` 2-D tensors. This product is efficiently computed
+    using the matrix chain order algorithm which selects the order in which incurs the lowest cost in terms
+    of arithmetic operations (`[CLRS]`_). Note that since is a function to compute the product, :math:`N`
+    needs to be greater than or equal to 2; if equal to 2 then a trivial matrix-matrix product is returned.
```

Comment:
I think the `N=2` case is straightforward. Maybe worth talking about `N=1` behavior?

Reply:
I thought it was evident that it would error out. I'll add a note about this as well.

Reply:
Oh if you are removing the check below, I think it's fine as-is. I was just confused by that check.

However, maybe reasonable to make `chain_matmul(A)` just return `A`?

Reply:
Sure, I will include that behaviour and add a note too.

Reply:
Thanks! I will need to read the literature a bit before commenting on the algorithm impl.